tests/test_matcher.py

```diff
@@ -161,3 +161,16 @@ def test_match_all():
     assert list(matcher.match_all("http://example.com/products")) == [1]
     assert list(matcher.match_all("http://foo.example.com/products")) == [2, 1]
     assert list(matcher.match_all("http://bar.example.com/products")) == [3, 4, 1]
+
+
+def test_include_universal():
+    matcher = URLMatcher()
+    matcher.add_or_update(1, Patterns(include=["example.com"]))
+    matcher.add_or_update(2, Patterns(include=[]))
```

Comment:
Could we also add another one with `Patterns(include=[""])`? 

just in case the definition of universal pattern changes later on and we can catch such regression